front-end/src/components/Balance.tsx

```diff
@@ -3,37 +3,36 @@
 // of the Apache-2.0 license. See the LICENSE file for details.
 
 import { DeriveBalancesAccount } from '@polkadot/api-derive/types';
+import { ApiPromiseContext } from '@substrate/context';
 import styled from '@xstyled/styled-components';
 import React, { useContext, useEffect, useState } from 'react';
 import formatBnBalance from 'src/util/formatBnBalance';
 
-import { ApiContext } from '../context/ApiContext';
-
 interface Props {
 	address: string
 	className?: string
 }
 
 const Balance = ({ address, className }: Props) => {
 	const [balance, setBalance] = useState<string>('0');
-	const { api } = useContext(ApiContext);
+	const { api, isApiReady } = useContext(ApiPromiseContext);
 
 	useEffect(() => {
-		let unsubscribe: () => void;
-
-		if (!api) {
-			console.error('polkadot/api not set');
+		if (!isApiReady) {
+			console.warn('Api not ready');
```

Comment:
These warning doesn't serve any purpose. unnecessarly bloat console errors. We only want the api ready log.

Reply:
agreed